lib/tuple.dart

```diff
@@ -66,7 +67,7 @@ class Tuple2<T1, T2> {
 
   @override
   bool operator ==(other) =>
-      other is Tuple2 && other.item1 == item1 && other.item2 == item2;
+      other is Tuple2<T1, T2> && other.item1 == item1 && other.item2 == item2;
```

Comment:
I'm not sure, but I think this might be a behavioral change for the package.

You might want to scope your PR to the original issue (the eqaulity fix), and the tests for that, and follow up with a 2nd PR with the additional changes. Someone else with more knowledge of the package can then review that PR.